src/noit_message_decoder.c

```diff
@@ -413,9 +413,10 @@ noit_metric_tags_count(const char *str, size_t strlen) {
   return rval;
 }
 
-const char *
-noit_metric_tags_parse_one(const char *tagnm, size_t tagnmlen,
-                           noit_metric_tag_t *output, mtev_boolean *toolong) {
+const char *noit_metric_tags_parse_one(const char *tagnm, size_t tagnmlen,
+                                       noit_metric_tag_t *output,
+                                       mtev_boolean *toolong,
+                                       int max_tag_length) {
```

Comment:
```suggestion
                                       const int32_t max_tag_length) {
```

Reply:
is `int32_t` to ensure that this behaves correctly on both 32 and 64 bit machines? are there any other reasons to use `int32_t` over `int`?
should I make this `uint32_t` while I'm at it since it should never be negative?

Reply:
> is `int32_t` to ensure that this behaves correctly on both 32 and 64 bit machines? are there any other reasons to use `int32_t` over `int`? 

https://www.google.com/search?q=why+prefer+int32_t+in+c&oq=why+prefer+int32_t+&aqs=chrome.1.69i57j33i160l4.8791j0j7&sourceid=chrome&ie=UTF-8

> should I make this `uint32_t` while I'm at it since it should never be negative?

Prefer signed to unsigned in C unless you're sure you know all the problems that come with ``unsigned`` are... with the exception of ``site_t`` or any other non-arithmetic use.